src/language-js/utils.js

```diff
@@ -1337,6 +1333,18 @@ function getComments(node, flags, fn) {
 const isNextLineEmpty = (node, { originalText }) =>
   isNextLineEmptyAfterIndex(originalText, locEnd(node));
 
+function isCallLikeExpression(node) {
+  return isCallOrOptionalCallExpression(node) || node.type === "NewExpression";
```

Comment:
I know this is silly, but can we add `import()` too?

Reply:
**Prettier pr-9992**
[Playground link](https://deploy-preview-9992--prettier.netlify.app/playground/#N4Igxg9gdgLgprEAuEBLAtgBwgJxgCgAJ8BDHAcwFd0EYBGASkIF4A+AHSkO+LKptgAmJm049eFarQDMIjlx6lJAmABY5YxXymwArBoXcl-WgDYD44zpgB2C1uW0AHPaPaVATlcSTsOgAZvKxU6RhZ5S3daOmFwwmBNcUgoAGcYQgAzCAgWQnYQLIh8gG5Enhw4GEocLkLCAGo8kAAjMhKygF8GYpAAGhAITBhUaBTkUDIcCAB3AAUyBDGUEgAbaZIATzH+5pwSMABrSoBlTH3UKHJkGBxKOH64dGa4ABMX14AZEkvKEnI4ABiuHQJBgw0uyBAJEoMAgfRAAAsYOgVgB1BGoeApM5gODHRaY1AAN0xG0hYBS2zQqTgeFme3IIOQGVWKXuIAAVikAB4AIT2hxOJBoHwucGZrPZXO5xwu5BWcAAipQIPAJSs2f0zjg2ThIa1nit4ZgcBcYKjUC8YAjkE5-FqpmzUXtMJDVustvCAI4q+D0wZLKEpAC0UDgr1e8IqPtQFXpfyZSBZGvZbPQqGut1TcoVytV4qTkv6MBIzQtVptSEExb2qBWcoAwhB0ImQHAUrp4ZQ2QAVUtLZOakBEu4ASSg71gxzApqGAEEJ8cYBsFeq2R0OkA)
```sh
--parser babel
--prose-wrap always
```

**Input:**
```jsx
import( (argument1) =>
    (argument2) =>
    (argument3) =>
    (argument4) =>
    (argument5) =>
    (argument6) =>
    (argument7) =>
    (argument8) =>
    (argument9) =>
    (argument10) =>
    (argument11) =>
    (argument12) => {
      const foo = "foo";
      return foo + "bar";
    });
```

**Output:**
```jsx
import(
  (argument1) =>
    (argument2) =>
    (argument3) =>
    (argument4) =>
    (argument5) =>
    (argument6) =>
    (argument7) =>
    (argument8) =>
    (argument9) =>
    (argument10) =>
    (argument11) =>
    (argument12) => {
      const foo = "foo";
      return foo + "bar";
    }
);

```

**Prettier pr-9992**
[Playground link](https://deploy-preview-9992--prettier.netlify.app/playground/#N4Igxg9gdgLgprEAuEBLAtgBwgJxgCgAJ8BDHAcwFd0EYBGASkIF4A+AHSkO+LKptgAmJm049eFarQDMIjlx6lJAmABY5YxXymwArBoXcl-WgDYD44zpgB2C1uW0AHPaPaVATlcSTsOgAZvKxU6RhZ5S3daOmFwwmBNcUgoAGcYQgAzCAgWQnYQLIh8gG5Enhw4GEocLkLCAGo8kAAjMhKygF8GYpAAGhAITBhUaBTkUDIcCAB3AAUyBDGUEgAbaZIATzH+5pwSMABrSoBlTH3UKHJkGBxKOH64dGa4ABMX14AZEkvKEnI4ABiuHQJBgw0uyBAJEoMAgfRAAAsYOgVgB1BGoeApM5gODHRaY1AAN0xG0hYBS2zQqTgeFme3IIOQGVWKXuIAAVikAB4AIT2hxOJBoHwucGZrPZXO5xwu5BWcAAipQIPAJSs2f0zjg2ThITQcKgNiQEfDMIbYKjUC8YKakE5-FqpmzUXtMJDVustvCAI4q+D0wZLKEpAC0UDgr1e8IqftQFXpfyZSBZGvZbPQqGut3TcoVytV4pTkv6MBIzStNrtglLe1QKzlAGEIOhkyA4CldPDKGyACrlpapzUgIl3ACSUHesGOYENQwAgpPjjANgr1WyOh0gA)
```sh
--parser meriyah
--prose-wrap always
```

**Input:**
```meriyah
import( (argument1) =>
    (argument2) =>
    (argument3) =>
    (argument4) =>
    (argument5) =>
    (argument6) =>
    (argument7) =>
    (argument8) =>
    (argument9) =>
    (argument10) =>
    (argument11) =>
    (argument12) => {
      const foo = "foo";
      return foo + "bar";
    });
```

**Output:**
```meriyah
import(
  (argument1) =>
  (argument2) =>
  (argument3) =>
  (argument4) =>
  (argument5) =>
  (argument6) =>
  (argument7) =>
  (argument8) =>
  (argument9) =>
  (argument10) =>
  (argument11) =>
  (argument12) => {
    const foo = "foo";
    return foo + "bar";
  }
);

```

Reply:
We cannot add `ImportExpression` to `isCallLikeExpression` because `import(...)` isn't like call expression. But I changed to format import and curried functions in the same way as call expressions ( 2de96ba ).

Reply:
Sorry, didn't see this, I think we should, we already treat it like `CallExpression` everywhere.

Reply:
What do you think about https://github.com/prettier/prettier/pull/9992#discussion_r558318092 ?

Reply:
This why I added https://github.com/prettier/prettier/blob/2f3fc241f8cb1867a0a3073ceac9d662e4b25fab/src/language-js/utils.js#L1210-L1235, should not use `node.arguments`.

Reply:
Okay, I'll fix.

Reply:
done 6fb35ae